src/popup/components/ConnectWalletForm.tsx

```diff
@@ -62,34 +65,34 @@ export const ConnectWalletForm = ({
   const [recurring, setRecurring] = React.useState<Inputs['recurring']>(
     defaultValues.recurring || false,
   );
+
   const [autoKeyShareFailed, setAutoKeyShareFailed] = React.useState(
     isAutoKeyAddFailed(state),
   );
 
   const resetState = React.useCallback(async () => {
     await clearConnectState();
-    setErrors((_) => ({ ..._, keyPair: '', connect: '' }));
+    setErrors((_) => ({ ..._, keyPair: null, connect: null }));
     setAutoKeyShareFailed(false);
   }, [clearConnectState]);
 
+  const toErrorInfo = React.useCallback(
+    (err?: string | ErrorWithKeyLike | null): ErrorInfo | null => {
```

Comment:
Why do we have the error as string as well? Can't we pass the whole error directly? For example here:

https://github.com/interledger/web-monetization-extension/blob/a012ce4ec04def3ee3a9400eab2f3b2411246f72/src/popup/components/ConnectWalletForm.tsx#L114-L124

Reply:
If we allow passing `{ message: string }` (which matches `Error`), we've to always add a check for `isErrorWithKey`, which is more expensive then `typeof err === 'string'`.

I'll look into rationalizing error types as a follow-up. I basically want only `Error` (`{ messsage: string }`) and `ErrorWithKeyLike`, with `failure` or `ErrorResponse` allowing returning serialization of either.